client/core/core.go

```diff
@@ -932,10 +937,35 @@ func (dc *dexConnection) reconcileTrades(srvOrderStatuses []*msgjson.OrderStatus
 			var oid order.OrderID
 			copy(oid[:], srvOrderStatus.ID)
 			trade := knownActiveTrades[oid] // no need to lock dc.tradeMtx
+			if trade == nil {
+				dc.log.Warnf("Server reported status for order %v that we did not request.", oid)
```

Comment:
This is a defense against a hostile or buggy server, which could cause client to panic.